base/random/RNGs.jl

```diff
@@ -328,8 +328,9 @@ function rand!(r::MersenneTwister, A::Union{Array{Float16},Array{Float32}},
     T = eltype(A)
     n = length(A)
     n128 = n * sizeof(T) ÷ 16
-    rand!(r, unsafe_wrap(Array, convert(Ptr{Float64}, pointer(A)), 2*n128),
-          2*n128, Close1Open2())
+    Base.@gc_preserve A rand!(r, unsafe_wrap(Array, convert(Ptr{Float64}, pointer(A)), 2*n128),
+                              2*n128, Close1Open2())
+    # FIXME: This code is completely invalid!!!
     A128 = unsafe_wrap(Array, convert(Ptr{UInt128}, pointer(A)), n128)
```

Comment:
I had plans to change this to avoid using `unsafe_wrap` for performance reasons (cf. #9174), but maybe I need to understand what is invalid here? Preserving `A` wouldn't be enough?

Reply:
You can't pass a managed pointer to `unsafe_wrap`.

Reply:
Ok thanks for the precision (this limitation is not documented in the docstring). I will work on fixing those cases.